### PR TITLE
tests: cleanup tests

### PR DESCRIPTION
**Pull request by mxschmitt**
Applied:
- https://github.com/microsoft/playwright/pull/3742
- https://github.com/microsoft/playwright/pull/6444

First iteration for reducing flaky tests, will follow up with more.

### CODE REVIEW

File: playwright/_impl/_connection.py [hunk 1]
```diff
@@ -338,7 +338,3 @@ def serialize_call_stack(stack_trace: traceback.StackSummary) -> List[Dict]:
         )
     stack.reverse()
     return stack
-
-
-def capture_call_stack() -> List[Dict]:
```
Comment (mxschmitt) [inline playwright/_impl/_connection.py:343]: this was unused

File: tests/async/test_browsertype_connect_cdp.py [hunk 2]
```diff
@@ -30,7 +30,6 @@ async def test_connect_to_an_existing_cdp_session(
     browser_server = await browser_type.launch(
         **launch_arguments, args=[f"--remote-debugging-port={port}"]
     )
-    wait_for_port(port)
```
Comment (mxschmitt) [inline tests/async/test_browsertype_connect_cdp.py:33]: We upstream wait now if the --remote-debugging-port arg is passed until the websocket server is ready.

File: tests/async/test_tracing.py [hunk 3]
```diff
@@ -28,6 +28,7 @@ async def test_browser_context_output_trace(
     page = await context.new_page()
     await page.goto(server.PREFIX + "/grid.html")
     await context.tracing.stop()
+    await page.wait_for_timeout(100)
```
Comment (dgozman) [inline tests/async/test_tracing.py:31]: Do we have this upstream? Perhaps make it 1000?

**Review by dgozman — APPROVED** [review]

File: tests/async/test_tracing.py (hunk 3 above)
Comment (mxschmitt) [reply tests/async/test_tracing.py]: I tried to repro it upstream, but wasn't successful with it, will investigate further in a follow-up.

**Pull request merged**